extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -197,7 +197,7 @@ protected RecordSupplier<String, String, ByteEntity> setupRecordSupplier() throw
             ioConfig.getAwsAssumedRoleArn(),
             ioConfig.getAwsExternalId()
         ),
-        ioConfig.getRecordsPerFetch(),
+        0, // no records-per-fetch, it is not used
```

Comment:
Maybe rephrase the comment for clarity:
```suggestion
        0, // need not specify recordsPerFetch as supervisor-side supplier does not fetch records
```